src/main/scanLogic/scanRunners/applicabilityScan.ts

```diff
@@ -97,25 +98,27 @@ export class ApplicabilityRunner extends BinaryRunner {
         if (!run) {
             return {} as ApplicabilityScanResponse;
         }
-        // Store all the rules that the run checked
-        let response: ApplicabilityScanResponse = {
-            scannedCve: run.tool.driver.rules?.map(rule => this.getCveFromRuleId(rule.id))
-        } as ApplicabilityScanResponse;
-
-        // Generate applicable data
         let applicable: Map<string, CveApplicableDetails> = new Map<string, CveApplicableDetails>();
+        let rules: AnalyzerRule[] = run.tool.driver.rules;
         let issues: AnalyzeIssue[] = run.results;
         if (issues) {
+            // Generate applicable data for all the issues
             issues.forEach(analyzeIssue => {
-                let applicableDetails: CveApplicableDetails = this.getOrCreateApplicableDetails(response.scannedCve, applicable, analyzeIssue);
+                let applicableDetails: CveApplicableDetails = this.getOrCreateApplicableDetails(
+                    analyzeIssue,
+                    applicable,
+                    rules.find(rule => rule.id === analyzeIssue.ruleId)?.fullDescription.text
```

Comment:
Let's use map here for better performance

Reply:
done